server/app/tasks/DatabaseSeedTask.java

```diff
@@ -0,0 +1,162 @@
+                      Lang.forCode("tl").toLocale(),
+                          "Pangalan (unang pangalan at ang unang titik ng apilyedo ay okay)",
+                      Lang.forCode("vi").toLocale(), "Tên (tên và họ viết tắt đều được)",
+                      Lang.forCode("en-US").toLocale(), "Please enter your first and last name",
+                      Lang.forCode("es-US").toLocale(),
+                          "Nombre (nombre y la inicial del apellido está bien)",
+                      Lang.forCode("zh-TW").toLocale(), "姓名（名字和姓氏第一個字母便可）")),
+              /* questionHelpText= */ LocalizedStrings.empty()),
+          new DateQuestionDefinition(
+              /* name= */ "Applicant Date of Birth",
+              /* enumeratorId= */ Optional.empty(),
+              /* description= */ "Applicant's date of birth",
+              /* questionText= */ LocalizedStrings.of(
+                  Lang.forCode("en-US").toLocale(),
+                  "Please enter your date of birth in the format mm/dd/yyyy"),
+              /* questionHelpText= */ LocalizedStrings.empty()));
+
+  private final Provider<QuestionService> questionServiceProvider;
+  private final Provider<VersionRepository> versionRepository;
+  private final Database database;
+
+  @Inject
+  public DatabaseSeedTask(
+      Provider<QuestionService> questionService, Provider<VersionRepository> versionRepository) {
+    this.questionServiceProvider = checkNotNull(questionService);
+    this.versionRepository = checkNotNull(versionRepository);
+    this.database = DB.getDefault();
+
+    this.run();
+  }
+
+  private void run() {
+    if (isMissingCanonicalQuestions()) {
+      // Ensure a draft version exists to avoid transaction collisions.
+      versionRepository.get().getDraftVersion();
+      inSerializableTransaction(this::seedCanonicalQuestions, 1);
+    }
+  }
+
+  /**
+   * Ensures that questions with names matching those in {@code CANONICAL_QUESTIONS} are present in
+   * the database, inserting the definitions in {@code CANONICAL_QUESTIONS} if any aren't found.
+   */
+  private void seedCanonicalQuestions() {
+    var questionService = questionServiceProvider.get();
+    ImmutableSet<String> existingQuestionNames = questionService.getQuestionNames();
+
+    for (QuestionDefinition questionDefinition : CANONICAL_QUESTIONS) {
+      if (existingQuestionNames.contains(questionDefinition.getName())) {
+        LOGGER.info(
+            "Canonical question \"%s\" exists at server start", questionDefinition.getName());
+        continue;
+      }
+
+      ErrorAnd<QuestionDefinition, CiviFormError> result =
+          questionService.create(questionDefinition);
+
+      if (result.isError()) {
+        String errorMessages =
+            result.getErrors().stream()
+                .map(CiviFormError::message)
+                .collect(Collectors.joining(", "));
+
+        LOGGER.error(
```

Comment:
What should we do in this case since downstream functionality will depend on it? Check-fail the server? Continue on and accept that downstream features that depend on these questions existing will throw errors?

Reply:
I think it should continue on after failing. This code will only run when the server starts for the first time with a completely fresh database. I'm leery of designing it in such a way that it could prevent the server from operating at all if something goes wrong.